python/ray/dashboard/modules/job/utils.py

```diff
@@ -303,3 +303,74 @@ async def find_jobs_by_job_ids(
             for job_info, submission_id in zip(job_infos, job_submission_ids)
         },
     }
+
+
+def fast_tail_last_n_lines(
+    path: str,
+    num_lines: int,
+    max_chars: int = 20000,
```

Comment:
I don't think `max_chars` should have a default value. Instead, require the caller to pass it